addrs.go

```diff
@@ -158,3 +160,29 @@ func ParseNetIP(s string) (ip net.IP, err error) {
 
 	return asNetIP(addr.Unmap()), nil
 }
+
+// ParsePort validates a string as a TCP port
+func ParsePort(s string) (port uint16, err error) {
+	if s == "" {
+		return 0, nil
+	}
+
+	prt, err := strconv.ParseInt(s, 10, 64)
```

Comment:
wouldn't be simpler using `strconv.ParseUint` to validate the range?

Reply:
I'm all for core.ParsePort(). we already have many pieces like this:
https://github.com/darvaza-proxy/core/blob/main/splithostport.go#L103
```go
func validPort(s string) bool {
	_, err := strconv.ParseUint(s, 10, 16)
	return err == nil
}
```